controller/api/serializers.py

```diff
@@ -35,6 +35,42 @@ def to_internal_value(self, data):
         return val
 
 
+class JSONIntFieldSerializer(serializers.Field):
```

Comment:
There should be python docstrings for these classes and methods. Also, I think subclassing `JSONFieldSerializer` could provide more code reuse. Something like this (untested!):
```python
class JSONFieldSerializer(serializers.Field):
    """
    A Django REST framework serializer for JSON data.
    """

    def to_representation(self, obj):
        """Serialize the field's JSON data, for read operations."""
        return obj

    def to_internal_value(self, data):
        """Deserialize the field's JSON data, for write operations."""
        Transform lists of primitive values dicts of native values to
        try:
            val = json.loads(data)
        except TypeError:
            val = data
        return val


class JSONIntFieldSerializer(JSONFieldSerializer):
    """
    A JSON serializer that coerces its data to integers.
    """

    def to_internal_value(self, data):
        """Deserialize the field's JSON integer data."""
        field = super(JSONIntFieldSerializer, self).to_internal_value(data)

        for k, v in field.viewitems():
            if v is not None:  # NoneType is used to unset a value
                field[k] = int(v)

        return field


class JSONStringFieldSerializer(JSONFieldSerializer):
    """
    A JSON serializer that coerces its data to strings.
    """

    def to_internal_value(self, data):
        """Deserialize the field's JSON string data."""
        field = super(JSONStringFieldSerializer, self).to_internal_value(data)

        for k, v in field.viewitems():
            if v is not None:  # NoneType is used to unset a value
                field[k] = unicode(v)

        return field
```